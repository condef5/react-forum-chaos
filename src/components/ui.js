/** @jsx jsx */
import { jsx } from "@emotion/core";

const styleInput = {
  background: "none",
  border: "1px solid #eaeaea",
  borderRadius: ".25rem",
  boxSizing: "border-box",
  display: "block",
  fontSize: "1rem",
  padding: ".5rem",
  width: "100%",
  "&:focus": {
    outline: "none",
    borderColor: "#444444"
  }
};

function Button({ styles, ...props }) {
  return (
    <button
      {...props}
      css={{
        backgroundColor: "black",
        border: "1px solid black",
        borderRadius: ".25rem",
        color: "white",
        cursor: "pointer",
        fontSize: ".8rem",
        padding: ".75rem 0",
        transition: "all 200ms ease",
        textAlign: "center",
        textTransform: "uppercase",
        width: "100%",
        "&:hover": {
          backgroundColor: "white",
          color: "black"
        },
        ...styles
      }}
    />
  );
}

function Card({ styles, ...props }) {
  return (
    <div
      {...props}
      css={{
        background: "white",
        borderRadius: ".5em",
        boxShadow: "0px 2px 5px 0px rgba(0, 0, 0, .12)",
        boxSizing: "border-box",
        padding: "2rem",
        width: "100%",
        ...styles
      }}
    />
  );
}

function Input({ styles, ...props }) {
  return (
    <input
      {...props}
      css={{
        ...styleInput,
        ...styles
      }}
    />
  );
}

function TextArea({ styles, ...props }) {
  return <textarea {...props} css={{ ...styleInput, ...styles }} />;
}

export { Button, Card, Input, TextArea };
