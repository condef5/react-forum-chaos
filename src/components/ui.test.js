import React from "react";
import { render } from "@testing-library/react";
import { Button, Card, Input, TextArea } from "./ui";

test("Button", () => {
  expect(
    render(<Button>This is a button</Button>).asFragment()
  ).toMatchSnapshot();
});

test("Card", () => {
  expect(render(<Card>This is card</Card>).asFragment()).toMatchSnapshot();
});

test("Input", () => {
  expect(render(<Input />).asFragment()).toMatchSnapshot();
});

test("TextArea", () => {
  expect(render(<TextArea />).asFragment()).toMatchSnapshot();
});
