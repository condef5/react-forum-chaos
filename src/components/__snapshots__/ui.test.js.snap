// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #f60;
  border: 1px solid #f60;
  border-radius: .25rem;
  color: white;
  cursor: pointer;
  font-size: .8rem;
  padding: .75rem 0;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  text-align: center;
  text-transform: uppercase;
  width: 100%;
}

.emotion-0:hover {
  background-color: white;
  color: #f60;
}

<button
    class="emotion-0"
  >
    This is a button
  </button>
</DocumentFragment>
`;

exports[`Card 1`] = `
<DocumentFragment>
  .emotion-0 {
  background: white;
  border-radius: .5em;
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,.12);
  box-sizing: border-box;
  padding: 2rem;
  width: 100%;
}

<div
    class="emotion-0"
  >
    This is card
  </div>
</DocumentFragment>
`;

exports[`Input 1`] = `
<DocumentFragment>
  .emotion-0 {
  background: none;
  border: 1px solid #eaeaea;
  border-radius: .25rem;
  box-sizing: border-box;
  display: block;
  font-size: 1rem;
  padding: .5rem;
  color: #34495e;
  width: 100%;
}

.emotion-0:focus {
  outline: none;
  border-color: rgba(255,102,0,0.5);
}

<input
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`TextArea 1`] = `
<DocumentFragment>
  .emotion-0 {
  background: none;
  border: 1px solid #eaeaea;
  border-radius: .25rem;
  box-sizing: border-box;
  display: block;
  font-size: 1rem;
  padding: .5rem;
  color: #34495e;
  width: 100%;
}

.emotion-0:focus {
  outline: none;
  border-color: rgba(255,102,0,0.5);
}

<textarea
    class="emotion-0"
  />
</DocumentFragment>
`;
